python/cudf/cudf/core/dataframe.py

```diff
@@ -405,12 +405,12 @@ def _setitem_tuple_arg(self, key, value):
                 value = as_column(value, length=length)
 
             new_col = cudf.Series(value, index=idx)
-            if not self._frame.empty:
+            if not (self._frame.empty and len(self._frame.index) == 0):
```

Comment:
Should the `empty` check just be replaced with your new check?

Reply:
Yup! Done 👍